pkg/ebpf/map.go

```diff
@@ -67,6 +67,9 @@ func (m *Map) OpenOrCreate() error {
 		PinPath: bpf.MapPrefixPath(),
 	}
 
+	mapType := bpf.GetMapType(bpf.MapType(m.spec.Type))
+	m.spec.Flags = m.spec.Flags | bpf.GetPreAllocateMapFlags(mapType)
```

Comment:
Does the `Flags` field hold other flags besides `BPF_F_NO_PREALLOC`?
Could it in the future?

Reply:
There are multiple possible other flags, this is just handling the "prealloc" case which has specific requirements depending on the map type (LPM cannot be preallocated; LRU for instance *must* be preallocated)